lib/spack/spack/main.py

```diff
@@ -856,9 +859,22 @@ def _main(argv=None):
     cmd_name = args.command[0]
     cmd_name = aliases.get(cmd_name, cmd_name)
 
-    command = parser.add_command(cmd_name)
+    # set up a bootstrap context, if asked.
+    # bootstrap context needs to include parsing the command, b/c things
+    # like `ConstraintAction` and `ConfigSetAction` happen at parse time.
+    bootstrap_context = llnl.util.lang.nullcontext()
+    if args.bootstrap:
+        import spack.bootstrap as bootstrap  # avoid circular imports
+        bootstrap_context = bootstrap.ensure_bootstrap_configuration()
+
+    with bootstrap_context:
+        finish_parse_and_run(parser, cmd_name, env_format_error)
```

Comment:
@tgamblin @alalazo Any chance this should be a `return` statement?  I'm suspicious because suddenly returning a value from `ci_rebuild` isn't resulting in that value turning up as the exit code of the spack command.

@zackgalbreath fyi

Reply:
@scottwittenburg I think you're right, looking at the previous code it seems we should return

Reply:
Ok, I can make a PR (with a test to make sure we notice in the future).

Reply:
It most definitely should be a return -- nice catch.